cmd/root.go

```diff
@@ -41,22 +40,9 @@ It can be used to export, import or sync entities to Kong.`,
 // sflags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
-	var wg sync.WaitGroup
-	var err error
-	const threads = 2
-	wg.Add(threads)
-
-	go func() {
-		sendAnalytics()
-		wg.Done()
-	}()
-
-	go func() {
-		err = rootCmd.Execute()
-		wg.Done()
-	}()
-
-	wg.Wait()
+	go sendAnalytics()
+
+	err := rootCmd.Execute()
```

Comment:
I think that we should do it by leaving the `WaitGroup` as it was, but passing a cancellable context to `sendAnalytics` and cancelling it after `rootCmd.Execute()` finishes.

Otherwise you cannot guarantee clean execution (for example, honoring `defer`red calls)

Reply:
I've updated the code to use context.

Are there any advantages to that approach other than clean execution (which doesn't matter here)?

Reply:
Discussed out of band. Recording the conclusions for future reference:
1. Running a goroutine that has a possibility of outliving `main()` results in an unpredictable, abrupt shutdown of that goroutine. This leads to surprising interruption of execution, and no guarantee of execution of `defer` calls.
2. If `main()` gets refactored in the future (and e.g. put in a library), then return from the function containing this snippet won't necessarily cause program shutdown, and thus the spawned goroutine may continue to run indefinitely.
3. `context` is the solution of choice for interruption of operations (especially HTTP calls).